tornjak-frontend/src/components/cluster-edit.tsx

```diff
@@ -0,0 +1,547 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import axios from 'axios';
+import { Dropdown, TextInput, FilterableMultiSelect, TextArea } from 'carbon-components-react';
+import GetApiServerUri from './helpers';
+import IsManager from './is_manager';
+import TornjakApi from './tornjak-api-helpers';
+import './style.css';
+import {
+  clustersListUpdateFunc,
+  clusterTypeInfoFunc,
+  serverSelectedFunc,
+  selectorInfoFunc,
+  agentsListUpdateFunc,
+  tornjakMessageFunc,
+  tornjakServerInfoUpdateFunc,
+  serverInfoUpdateFunc
+} from 'redux/actions';
+import { RootState } from 'redux/reducers';
+import {
+  AgentsList,
+  ClustersList,
+  ServerInfo
+} from './types'
+
+type ClusterEditProp = {
+  // dispatches a payload for list of clusters with their metadata info as an array of ClustersList Type and has a return type of void
+  clustersListUpdateFunc: (globalClustersList: ClustersList[]) => void,
+  // dispatches a payload for the tornjak error messsege and has a return type of void
+  tornjakMessageFunc: (globalErrorMessage: string) => void,
+  // cluster types as array of strings
+  clusterTypeList: string[],
+  // the selected server for manager mode 
+  globalServerSelected: string,
+  // tornjak error messege
+  globalErrorMessage: string,
+  // list of clusters with their metadata info as an array of ClustersList Type or can be undefined if no array present
+  globalClustersList: ClustersList[] | undefined,
```

Comment:
is there a way we can remove the or for undefined here? Is there a particular use-case around it or is it that we have to initialize on creation - i.e. in the redux/helper fns? (same question for all types initialized this way).

Reply:
Removed!